### PR TITLE
Multideterminant wave functions

### PR DESCRIPTION
**Pull request by sapatha2**
Coded up multislater wave functions for pyqmc. Test method at end of file passes gradients, pgradients, laplacian, and update internals. VMC energies are tested as well and are consistent with PySCF CASCI calculations.

### CODE REVIEW

**Comment by sapatha2** [comment]
Moved testing file to test/ directory, everything seems to work correctly now!

File: pyqmc/multislater.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
+import numpy as np
+from slateruhf import sherman_morrison_row
+
+#Taken from autogenv2/pyscf2qwalk
```
Comment (lkwagner) [inline pyqmc/multislater.py]: Remove this comment; it's not necessary.

File: pyqmc/multislater.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+def binary_to_occ(S, ncore):
```
Comment (lkwagner) [inline pyqmc/multislater.py]: need a docstring; what are the inputs/outputs of this function?

File: pyqmc/multislater.py [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+        ratio /= (curr_val[0]*np.exp(curr_val[1]))
+        return ratio
+    
+    def pgradient(self):
+        """Compute the parameter gradient of Psi. 
+        Returns d_p \Psi/\Psi as a dictionary of numpy arrays,
+        which correspond to the parameter dictionary.
+        """
+        d = {}
+        for parm in self.parameters:
```
Comment (lkwagner) [inline pyqmc/multislater.py]: Why loop and check conditions? Why not just set d['det_coeff']?

File: tests/test_multislater.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        mc = mcscf.CASCI(mf,ncas=2,nelecas=(1,1))
+        mc.kernel()
+        wf = MultiSlater(mol, mf, mc)
+        
+        nconf = 10
+        nelec = np.sum(mol.nelec)
+        configs = np.random.randn(nconf, nelec, 3)
+        
+        print("Testing internals:", testwf.test_updateinternals(wf, configs))
+        for delta in [1e-3, 1e-4, 1e-5, 1e-6, 1e-7]:
```
Comment (lkwagner) [inline tests/test_multislater.py]: Need assert statements so the automated tests work.

File: tests/test_multislater.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+        warmup = 30
+        coords = initial_guess(mol, nconf)
+        df, coords = vmc(
+            wf, coords, nsteps=nsteps, accumulators={"energy": EnergyAccumulator(mol)}
+        )
+
+        df = pd.DataFrame(df)
+        en = np.mean(df["energytotal"][warmup:])
+        err = np.std(df["energytotal"][warmup:]) / np.sqrt(nsteps - warmup)
+        print('VMC E = ',en,'+/-',err)
```
Comment (lkwagner) [inline tests/test_multislater.py]: Need an assert here too.

File: pyqmc/multislater.py (hunk 2 above)
Comment (sapatha2) [reply pyqmc/multislater.py]: handled

**Comment by sapatha2** [comment]
Addressed comments from Lucas, ran pytest with no problems locally.

**Pull request merged**